src/common/cmdparse.cc

```diff
@@ -14,13 +14,37 @@
 
 #include <cxxabi.h>
 #include "common/cmdparse.h"
+#include "common/Formatter.h"
 #include "include/str_list.h"
 #include "json_spirit/json_spirit.h"
 #include "common/debug.h"
 
 using namespace std;
 
 /**
+ * Given a cmddesc like "foo baz name=bar,type=CephString",
+ * return the prefix "foo baz".
+ */
+std::string cmddesc_get_prefix(const std::string &cmddesc)
+{
+  stringstream ss(cmddesc);
+  std::string word;
+  std::ostringstream result;
+  bool first = true;
+  while (std::getline(ss, word, ' ')) {
+    if (word.find_first_of(",=") == string::npos) {
```

Comment:
could just break if `word.find_first_of(",=") != word.npos`. or we we can just 
```c++
return cmddesc.substr(0, cmddesc.rfind(' '));
```
assuming there is no space in the parameter list spec.